packages/entity-server/src/routes/Route.ts

```diff
@@ -5,14 +5,16 @@
 import { Request, NextFunction, Response } from 'express';
 import { respond } from '@tupaia/utils';
 
-export class Route {
-  req: Request;
+type ResBody<Type> = Type extends Response<infer Body> ? Body : null; // Infers body of response type
 
-  res: Response;
+export class Route<Req extends Request = Request, Res extends Response = Response> {
```

Comment:
Made the base `Route` class generic so that we can specify what Request/Response we expect from each route